src/etc/inc/dyndns.class

```diff
@@ -1117,7 +1122,92 @@
 				curl_setopt($ch, CURLOPT_HEADER, 1);
 				curl_setopt($ch, CURLOPT_CUSTOMREQUEST, 'PUT');
 				break;
+				case 'linode':
+				case 'linode-v6':
+					$linode_api = "https://api.linode.com/v4";
+					$record_type = $this->_useIPv6 ? "AAAA" : "A";
+					$record_name = $this->_dnsHost == "@" ? "" : $this->_dnsHost;
+					$err_prefix = gettext("Dynamic DNS") . " {$this->_dnsService} ({$this->_FQDN}): (" . gettext("Error") . ")";
+					curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
+					curl_setopt($ch, CURLOPT_HTTPHEADER, array(
+						'Accept: application/json',
+						'Content-Type: application/json',
+						'Authorization: Bearer ' . $this->_dnsPass
+					));
+
+					// get domain id
+					curl_setopt($ch, CURLOPT_URL, "$linode_api/domains");
+					$domains_output = curl_exec($ch);
+					$http_code = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+					if ( $http_code == 401 && preg_match('/invalid oauth token/i', $domains_output) ) {
+						log_error("$err_prefix " . gettext("querying domains") . ": " . gettext("User Authentication Failed"));
+						return false;
+					} else if ( $http_code == 401 ) {
+						log_error("$err_prefix " . gettext("querying domains") . ": " . gettext("User Authorization Failed"));
+						return false;
+					} else if ( $http_code != 200 ) {
+						log_error("$err_prefix " . gettext("querying domains") . ": " .
+							( isset($domains_result["errors"][0]["reason"]) ? $domains_result["errors"][0]["reason"] : "HTTP $http_code" ) );
+						return false;
+					}
+
+					$domains_result = json_decode($domains_output, TRUE);
+					foreach($domains_result["data"] as $domains_entry) {
+						if ($domains_entry["domain"] == $this->_dnsDomain) {
+							$domain_id = $domains_entry["id"];
+						}
+					}
+					if ( ! $domain_id ) {
+						log_error("$err_prefix " . gettext("no domain ID for domain") . ": '{$this->_dnsDomain}'");
+						return false;
+					}
+					if ($this->_dnsVerboseLog) {
+						log_error("_update(): " . sprintf(gettext("found domain id: %s"), $domain_id));
+					}
+
+					// get existing record if present
+					curl_setopt($ch, CURLOPT_URL, "$linode_api/domains/$domain_id/records");
```

Comment:
Variable references in strings should have braces around them, e.g. `{$linode_api}` ( https://www.netgate.com/docs/pfsense/development/developer-style-guide.html#general-rules )